lib/codeowners/checker.rb

```diff
@@ -16,15 +16,18 @@ module Codeowners
   # By default (:validate_owners property) it also reads OWNERS with list of all
   # possible/valid owners and validates every owner in CODEOWNERS is defined in OWNERS
   class Checker
-    attr_reader :owners_list
+    attr_reader :owners_list, :enabled_checks
+
+    ALL_CHECKS = %i[missing_ref useless_pattern invalid_owner unrecognized_line].freeze
 
     # Get repo metadata and compare with the owners
-    def initialize(repo, from, to)
+    def initialize(repo, from, to, enabled_checks = ALL_CHECKS)
```

Comment:
with 4 args this screams "keyword arguments, please" :)

Reply:
Sure, that makes sense. Should I make all of them keyword arguments?

Reply:
I just did that way so it wasn't a breaking change